### PR TITLE
Rewrite GEN_KW unit tests to be more efficient

### PR DESCRIPTION
**Pull request by andreas-el**
**Issue**
Resolves #my_issue

**Approach**
_Short description of the approach_

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py [hunk 1]
```diff
@@ -108,6 +108,58 @@ def test_gen_kw(tmpdir, config_str, expected, extra_files, expectation):
...
+            does_not_raise(),
+            "somepath/",
+        ),
+        # This test was added to show current behaviour for Ert.
+        # If absolute paths should be possible to be used like this is up for debate.
+        (
+            "GEN_KW KW_NAME template.txt /tmp/somepath/kw.txt prior.txt INIT_FILES:custom_param%d",
+            "MY_KEYWORD 1.31",
+            [("custom_param0", "MY_KEYWORD 1.31")],
+            does_not_raise(),
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py:128]: Seing as both parameters are using `does_not_raise` it should be fine to remove `expectation` from the list

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py [hunk 2]
```diff
@@ -108,6 +108,58 @@ def test_gen_kw(tmpdir, config_str, expected, extra_files, expectation):
...
+            "GEN_KW KW_NAME template.txt somepath/kw.txt prior.txt\nRANDOM_SEED 1234",
+            "MY_KEYWORD -0.881423",
+            [],
+            does_not_raise(),
+            "somepath/",
+        ),
+        # This test was added to show current behaviour for Ert.
+        # If absolute paths should be possible to be used like this is up for debate.
+        (
+            "GEN_KW KW_NAME template.txt /tmp/somepath/kw.txt prior.txt INIT_FILES:custom_param%d",
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py:125]: Can probably simplify this, or is there a reason you would like to use `INIT_FILES`? Does not seem to be present in the test this is replacing? Is it correct that you want to test `somepath/` and `tmp/somepath`?

**Review by oyvindeide — COMMENTED** [review]
Thanks for the PR, some minor comments

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py [hunk 3]
```diff
@@ -108,6 +108,58 @@ def test_gen_kw(tmpdir, config_str, expected, extra_files, expectation):
...
+            write_file(fname, contents)
+
+        with expectation:
+            create_runpath("config.ert")
+            assert (
+                Path(
+                    "simulations/realization-0/iter-0/" + relpath + "kw.txt"
+                ).read_text(encoding="utf-8")
+                == expected
+            )
```
Comment (oyvindeide) [inline tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py:160]: Maybe something like this?

```suggestion
def test_gen_kw_outfile_will_use_paths(
    tmpdir, relpath
):
    with tmpdir.as_cwd():
        config = dedent(
            f"""
        JOBNAME my_name%d
        NUM_REALIZATIONS 1
        GEN_KW KW_NAME template.txt {relpath} prior.txt
        """
        )
        with open("config.ert", mode="w", encoding="utf-8") as fh:
            fh.writelines(config)
        with open("template.txt", mode="w", encoding="utf-8") as fh:
            fh.writelines("MY_KEYWORD <MY_KEYWORD>")
        with open("prior.txt", mode="w", encoding="utf-8") as fh:
            fh.writelines("MY_KEYWORD NORMAL 0 1")

         create_runpath("config.ert")
         assert (
                Path(
                    f"simulations/realization-0/iter-0/{relpath}kw.txt"
                ).read_text(encoding="utf-8")
                == expected
            )
```

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py (hunk 2 above)
Comment (andreas-el) [reply tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py]: Yes. The original issue was that ert did not store the files in the appropriate location. We need to verify that the file is at that specific place.

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py (hunk 1 above)
Comment (andreas-el) [reply tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py]: Correct. Is removed now.

File: tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py (hunk 3 above)
Comment (andreas-el) [reply tests/unit_tests/c_wrappers/integration/test_parameter_sample_types.py]: Great suggestion, have updated my code somewhat similar.

**Review by oyvindeide — APPROVED** [review]
LGTM!

**Pull request merged**